code/modules/power/supermatter/supermatter.dm

```diff
@@ -253,6 +278,39 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	if(final_countdown)
 		. += "casuality_field"
 
+	// Switch the distortion effect based on the crystal's state
+	var/current_status = get_status()
+	if (current_status != last_status)
```

Comment:
This check shouldn't be performed in here, as the overlays are cut when update_overlays is called. Inside of update_overlays you must always update the supermatter overlay, and instead only call update_icon when the current status is changed (Move the change check out of here and to the previous location where update_icon is called).

Reply:
Done